core/data/src/main/java/com/google/samples/apps/nowinandroid/core/data/repository/fake/FakeNewsRepository.kt

```diff
@@ -43,23 +44,25 @@ class FakeNewsRepository @Inject constructor(
     private val datasource: FakeNiaNetworkDataSource,
 ) : NewsRepository {
 
-    override fun getNewsResources(): Flow<List<NewsResource>> =
-        flow {
-            emit(
-                datasource.getNewsResources()
-                    .map(NetworkNewsResource::asEntity)
-                    .map(NewsResourceEntity::asExternalModel),
-            )
-        }.flowOn(ioDispatcher)
-
     override fun getNewsResources(
-        filterTopicIds: Set<String>,
+        query: NewsResourceQuery,
     ): Flow<List<NewsResource>> =
         flow {
             emit(
                 datasource
                     .getNewsResources()
-                    .filter { it.topics.intersect(filterTopicIds).isNotEmpty() }
+                    .filter { networkNewsResource ->
+                        listOfNotNull(
+                            // True by default
```

Comment:
I want to evaluate wether a news resource should be filtered out.

By default, the news resource should be kept so the first entry in the list is `true`. If the `filterTopicIds` is present, and the news resource doesn't match, the list becomes `[true, false]`, otherwise it's just `true`. The same goes for `filterNewsIds`. The filter lambda only admits the news resource if all items in the list are true.

This could be written more imperatively as

```
                  .filter { networkNewsResource ->
                     true &&
                     (query.filterTopicIds?.let { filterTopicIds ->
                              networkNewsResource.topics.intersect(filterTopicIds).isNotEmpty()
                      } ?: true) &&
                      (query.filterNewsIds?.let { filterNewsIds ->
                              filterNewsIds.contains(networkNewsResource.id)
                       } ?: true)
                  }
```

Reply:
Sorry, when I said "expand on this" I meant in the code comment, rather than here. Thanks for the explanation though. I'd suggest adding a comment along the lines of: 

// Filter out any news resources which don't match the current query. If no query parameters (filterTopicIds or filterNewsIds) are specified then the news resource is returned.

Reply:
Done.